src/net.rs

```diff
@@ -1,26 +1,30 @@
 use std::io::Write;
 use std::path::Path;
 
-use openssl::ssl::{SslContext, SslMethod};
+use openssl::ssl::{SslContext, Ssl, SslMethod, SslStream, SSL_VERIFY_PEER};
 use openssl::ssl::error::SslError;
 use openssl::x509::X509FileType;
 
-use std::sync::Arc;
+use openssl_verify;
 
-use hyper::net::{Openssl, HttpsConnector, Fresh};
+use hyper;
+use hyper::net::{HttpsConnector, Fresh, HttpStream};
 use hyper::method::Method;
 use hyper::client::{Client, RequestBuilder};
 use hyper::client::request::Request;
 
 use url::Url;
 
+pub struct OpensslClient(SslContext);
+
 pub fn ssl_context<C>(cacert: C,
                       cert: Option<C>,
                       key: Option<C>)
-                      -> Result<Openssl, SslError>
+                      -> Result<OpensslClient, SslError>
     where C: AsRef<Path>
 {
     let mut ctx = SslContext::new(SslMethod::Tlsv1_2).unwrap();
+    try!(ctx.set_default_verify_paths());
```

Comment:
We discussed this a little bit yesterday, but if we're going to include more rust projects in our shipped code (which I'm all about, rust seems awesome) we should have a leatherman-rust project that includes this security sensitive setup. We're going to the effort here to make sure that we have things configured correctly and securely, at some future date we should make sure that our efforts can be reused for other projects.

Not that we need to do this right now, of course - just seeding thoughts for the future.